src/state/slices/opportunitiesSlice/utils.ts

```diff
@@ -28,3 +29,6 @@ export const filterUserStakingIdByStakingIdCompareFn = (
 
   return deserializedStakingId === stakingId
 }
+
+export const toOpportunityId = (...[args]: Parameters<typeof toAssetId>) =>
```

Comment:
The second guts of this PR - all LpIds and StakingIds are AssetIds, but not all AssetIds are LpIds/StakingIds. By calling this in place of `toAssetId` when we need an opportunity ID, we slap the right dual flavor in

![image](https://user-images.githubusercontent.com/17035424/202209533-bbb3ef2c-0841-4d84-b2d8-0371fb35c1d1.png)